### ISSUE TITLE
Spamming notifications

### ISSUE THREAD

**Issue by flip111** [penguin-teal/gnome-openweather#32]
## Description

![image](https://github.com/penguin-teal/gnome-openweather/assets/2244480/0c190017-5d54-4bce-8bdd-0e3a779bb31e)

This notification keeps coming up every second.

## System

Ubuntu 24.04

## Settings

```json
{"app-version":"135","git-version":"135.r0.ge994aeb","gnome-version":"46.0","user-locale":"en","appid":"<KEY>","actual-city":"0","has-run":"true","decimal-places":"0","delay-ext-init":"1","locs":"[(0, '<NAME>', 0, '<PLACE>'), (0, '<NAME>', 0, '<PLACE>')]","my-loc-prov":"'ipinfoio'","weatherapidotcom-key":"<KEY>","show-sunsetrise-in-panel":"true","clock-format":"'24h'","menu-alignment":"73.0","position-in-panel":"'center'","show-comment-in-panel":"false"}
```

**Comment by furkansimsekli** [comment]
It's applicable for both of the providers, because the notifications kept coming even when I changed the provider.

**Comment by derek-shnosh** [comment]
I'm getting the same repeat notifications. This is potentially an API limit imposed by the providers. I'm using weatherapi.com and am going to sign up for my own [free API key](https://www.weatherapi.com/pricing.aspx) to see if this resolves the issue.

**Comment by fiftydinar** [comment]
What's interesting is that I don't have this issue in latest toppk fork version I use on another system.

**Comment by Sidethorn** [comment]
Same here from this morning

**Comment by w-j-r** [comment]
Using my own api key for openweathermap immediately resolved the issue. Make sure you switch on "Use custom api key" just above where you enter your personal key, I missed this at first.

**Comment by fiftydinar** [comment]
> Using my own api key for openweathermap immediately resolved the issue. Make sure you switch on "Use custom api key" just above where you enter your personal key, I missed this at first.

I confirm that this fixes the issue.

**Comment by penguin-teal** [comment]
Yes the notifications shouldn't spam, I just came to that issue upon power on to my system. I will fix that.

The reason this even happened is because there are too many users now.
I am in the process of adding another provider and will probably add more to attempt to balance the load out.
@fiftydinar the reason the toppk one works is because it uses a different API key and therefore isn't hitting the limit. This is effectively the same as using your own API key.

The surefire way to ensure you can always use the provider you want is to use your own custom key. I will do my best to alleviate the issues as much as I can though.

**Comment by mokilcde** [comment]
I had the same issue
When I am deactivate the extension and activate after a restart it works but I had the problem after hibernation or suspend to ram.

**Comment by federicolencina** [comment]
I found a temporal fix for this.
Use a Personal API Key. And the bug has gone.

**Comment by federicolencina** [comment]
I found a temporary fix for this.
Use a Personal API Key.
You can get your Personal API Key at: <https://openweathermap.org/api>

**Comment by robatino** [comment]
> The reason this even happened is because there are too many users now.

I'm not sure that's the actual cause - on my main machine, where I've been running 135 for some time, I've never seen this error even once. I didn't put in my personal API keys until a few days ago. I was seeing this error on another machine at the same time the main one was working normally. Anyway, I have personal API keys for both providers and am using them on all my machines now and haven't seen the problem so far.

**Comment by penguin-teal** [comment]
Kabloom
Fixed with ea129635966c97deed1bd37bb8f7eaca704da7f2

**Comment by flip111** [comment]
Thanks @penguin-teal could you make a new release so that the extension can be turned on again?

**Comment by penguin-teal** [comment]
@flip111 you should be able to turn the current version on as it's a new month and the rate limits have not been reached.
I want to finish up something in the new version (to hopefully alleviate more rate limiting issues) and am very busy but should be able to release it in a few days. I am adding "Visual Crossing" weather provider which supports 14-day-out forecasts.

**Comment by penguin-teal** [comment]
@BloodyIron is it spamming every second or just every so often?

**Comment by flip111** [comment]
Multiple times per day still

**Comment by BloodyIron** [comment]
> @BloodyIron is it spamming every second or just every so often?

I've honestly lost count of how many times per day, probably hundreds. It won't stop.

**Comment by BloodyIron** [comment]
So can we have this re-opened please?

**Comment by penguin-teal** [comment]
@BloodyIron I suppose the request here would be to send notifications less.

**Comment by BloodyIron** [comment]
> @BloodyIron I suppose the request here would be to send notifications less?

That would be half of it, but more some sort of solution whereby the default behaviour is it can work an entire month out of the box. Or maybe just make it broken by default forcing people to get API keys, I dunno. I would prefer to not really need to sign up for yet-another-account just to use this extension without being spammed by rate limit warnings.

Like, I'm not entirely seeing how what I'm seeking here is unclear. I installed an addon to see the weather, and it's in the end turned into spam messages about rate limits. How would you feel about that in my shoes? I bet you'd be annoyed.

**Comment by BloodyIron** [comment]
> @BloodyIron the alternative for the "not working" thing is in progress #67, essentially I was just saying that this issue ticket itself is a separate issue. The notifications I think are going to have to be a thing for certain providers, and the feedback on how to deliver those notifications is what I needed. I will probably end up changing to warn once on start-up and not again.

Ahh thanks for bringing that to my attention! Yay! Not sure what more to add that's helpful right now, sorry.

**Comment by derek-shnosh** [comment]
> @BloodyIron the alternative for the "not working" thing is in progress #67, essentially I was just saying that this issue ticket itself is a separate issue. The notifications I think are going to have to be a thing for certain providers, and the feedback on how to deliver those notifications is what I needed. I will probably end up changing to warn once on start-up and not again.

Maybe a subtle `!` or warning icon in the topbar to non-intrusively notify the user of an API key issue, where clicking on OpenWeather in the top bar would then have a message somewhere?

**Comment by penguin-teal** [comment]
@derek-shnosh I like that idea!

**Comment by BloodyIron** [comment]
> > @BloodyIron the alternative for the "not working" thing is in progress #67, essentially I was just saying that this issue ticket itself is a separate issue. The notifications I think are going to have to be a thing for certain providers, and the feedback on how to deliver those notifications is what I needed. I will probably end up changing to warn once on start-up and not again.
> 
> Maybe a subtle `!` or warning icon in the topbar to non-intrusively notify the user of an API key issue, where clicking on OpenWeather in the top bar would then have a message somewhere?

A shift in the kind of notification I think would be a worthwhile aspect, regardless of my other thoughts on "default behaviour" for working or not. I'd propose a careful balance between subtle, and not too subtle, heh. Maybe not red (so it's not tooooo scary), but also maybe not white/black (so it doesn't get ignored). Unsure if Yellow is a good idea, maybe Orange or Purple? Anyways, I do like that idea for that aspect!
